File: src/main/java/com/koddy/server/member/domain/model/Role.java
package com.koddy.server.member.domain.model;


import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum Role {
    MENTOR("ROLE_MENTOR", "멘토"),
    MENTEE("ROLE_MENTEE", "멘티"),
    ADMIN("ROLE_ADMIN", "관리자"),
    ;

    private final String authority;
    private final String value;

    public static class Value {
        public static final String MENTOR = "MENTOR";
        public static final String MENTEE = "MENTEE";
    }
}

File: src/main/java/com/koddy/server/member/domain/model/Role.kt
package com.koddy.server.member.domain.model

enum class Role(
    val authority: String,
    val description: String,
) {
    MENTOR("ROLE_MENTOR", "멘토"),
    MENTEE("ROLE_MENTEE", "멘티"),
    ADMIN("ROLE_ADMIN", "관리자"),
    ;

    companion object {
        const val MENTOR_VALUE = "MENTOR"
        const val MENTEE_VALUE = "MENTEE"
    }
}
